src/dos/program_more_output.cpp

```diff
@@ -699,41 +705,8 @@ std::string MoreOutputFiles::GetShortPath(const std::string &file_path,
 	// We need to make sure the path and file name fits within
 	// the designated space - if not, we have to shorten it.
 
-	// The shortest name we should be able to display is:
-	// - 3 dots (ellipsis)
-	// - 1 path separator
-	// - 8 characters of name
-	// - 1 dot
-	// - 3 characters of extension
-	// This gives 16 characters.
-	// We need to keep the last column free (reduces max length by 1).
-	// Format string contains '%s' (increases max length by 2).
-	constexpr size_t min = 16;
-	const auto max = GetMaxColumns() - std::strlen(MSG_Get(msg_id)) + 1;
-	const auto max_len = std::max(min, max);
-
-	// Nothing to do if file name matches the constraint
-	if (file_path.length() <= max_len) {
-		return file_path;
-	}
-
-	// We need to shorten the name - try to strip part of the path
-	static const std::string ellipsis = "...";
-	auto shortened = file_path;
-	while (shortened.length() > max_len &&
-	       std::count(shortened.begin(), shortened.end(), '\\') > 1) {
-		// Strip one level of path at a time
-		const auto pos = shortened.find('\\', shortened.find('\\') + 1);
-		shortened      = ellipsis + shortened.substr(pos);
-	}
-
-	// If still too long, just cut away the beginning
-	const auto len = shortened.length();
-	if (len > max_len) {
-		shortened = ellipsis + shortened.substr(len - max_len + 3);
-	}
-
-	return shortened;
+	const auto max_len = GetMaxColumns() - std::strlen(MSG_Get(msg_id)) + 1;
+	return shorten_path(file_path, max_len);
```

Comment:
That's a big improvement! :+1:

Reply:
Actually, most of the code is moved somewhere else, without many changes. I needed it to be reusable.

Reply:
(good to know; at least is make this area clearer)